internal/collector/signal/signal_test.go

```diff
@@ -0,0 +1,446 @@
+// Copyright 2022 Criticality Score Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signal
```

Comment:
Thanks for adding tests here.

Some general thoughts after looking through this PR.

- `iterSetFields(...)` never returns and error, nor do any of the callbacks. This functionality can be removed - this makes the need to test it directly basically unnecessary.
- `parseStructField()` and `iterSetFields()` can be tested indirectly through the public methods that use them. This makes it easier to refactor code without needing to also spend lots of time updating tests.
- `package signal` can be changed to `package signal_test` to require interacting with the public methods only.

Reply:
I have made some changes to the implementation so that it now uses public methods and the package name has been changed to 'signal_test'. Thank you for your help.